### PR TITLE
docs: add comprehensive contributing guidelines with TDD requirements

### PR DESCRIPTION
**Pull request by CurroRodriguez**
## Summary
Add detailed CONTRIBUTING.md file that includes:
- Development setup instructions with uv dependency manager
- Test-Driven Development (TDD) workflow and requirements
- Mandatory unit testing requirements with 75% minimum coverage
- Code quality standards and Python style guidelines
- Pull request process and commit message conventions
- Bug reporting and enhancement suggestion templates
- Documentation guidelines and community resources

Emphasizes TDD approach as a core requirement and makes it clear that all contributions must include appropriate unit tests. Links to uv installation page to support all operating systems.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,433 @@
...
+#### Test Coverage Standards
+
+- **Minimum coverage**: 75% (enforced by CI)
+- **Target coverage**: 85% or higher
+- **New code**: Must have 100% coverage for new modules/features
+- **Bug fixes**: Must include a test that reproduces the bug
+
+#### Writing Unit Tests
+
+Tests are located in the `test/` directory and use `pytest` and `assertpy`:
```
[the remark this answers is not in the document]
Comment (CurroRodriguez) [reply CONTRIBUTING.md]: I will keep this because I plan to migrate all the tests to more standard pytest.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,433 @@
...
+    assert result.success is True
+
+# 2. Write minimal code to pass (GREEN)
+class MyNewTask(Task):
+    def execute(self):
+        return Result(success=True)
+
+# 3. Refactor (REFACTOR)
+# Clean up, add error handling, improve names, etc.
+```
```
[the remark this answers is not in the document]
Comment (CurroRodriguez) [reply CONTRIBUTING.md]: @copilot open a new pull request to apply changes based on [this feedback](https://github.com/abantos/bolt/pull/129#discussion_r2530078459)

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,431 @@
...
+        # Act
+        result = task(config)
+        
+        # Assert
+        assert_that(result).is_not_none()
+        assert_that(result.success).is_true()
+        
+    def test_feature_handles_error_condition(self):
+        """Test error cases and edge conditions"""
+        with self.assertRaises(RequiredConfigurationError):
```
[the remark this answers is not in the document]
Comment (CurroRodriguez) [reply CONTRIBUTING.md]: @copilot open a new pull request to apply changes based on [this feedback](https://github.com/abantos/bolt/pull/129#discussion_r2530084952)

[the remark this answers is not in the document]
Comment (CurroRodriguez) [reply CONTRIBUTING.md]: @copilot open a new pull request to apply changes based on [this feedback](https://github.com/abantos/bolt/pull/129#discussion_r2530084952)
